bcda/main_test.go

```diff
@@ -640,52 +648,58 @@ func (s *MainTestSuite) TestCleanArchive() {
 
 func (s *MainTestSuite) TestRevokeToken() {
 
-	// init
-	s.SetupAuthBackend()
-
-	assert := assert.New(s.T())
-
-	// Create a token
-	tokenInfo, err := createAlphaToken(720, "Small")
-	assert.Nil(err)
-	checkTokenInfo(s, tokenInfo)
-	alphaTokenData := strings.Split(tokenInfo, "\n")
-	alphaTokenString := alphaTokenData[2]
-
-	// Create a token
-	userUUID := "82503A18-BF3B-436D-BA7B-BAE09B7FFD2F"
-	tokenString, err := createAccessToken(userUUID)
-	assert.Nil(err)
-
-	buf := new(bytes.Buffer)
-	s.testApp.Writer = buf
-
-	// Negative case - attempt to revoke a token passing in a blank token string
-	args := []string{"bcda", "revoke-token", "--access-token", ""}
-	err = s.testApp.Run(args)
-	assert.Equal("Access token (--access-token) must be provided", err.Error())
-	assert.Equal(0, buf.Len())
-	buf.Reset()
-
-	// Negative case - attempt to revoke a token passing in an invalid token string
-	args = []string{"bcda", "revoke-token", "--access-token", "abcdefg"}
-	err = s.testApp.Run(args)
-	assert.Equal("token contains an invalid number of segments", err.Error())
-	assert.Equal(0, buf.Len())
-	buf.Reset()
-
-	// Positive case - revoke a token passing in a valid token string (alpha)
-	args = []string{"bcda", "revoke-token", "--access-token", alphaTokenString}
-	err = s.testApp.Run(args)
-	assert.Nil(err)
-	assert.Equal("Access token has been deactivated\n", buf.String())
-	buf.Reset()
+	switch auth.GetProvider().(type) {
```

Comment:
originalAuthProvider := auth.GetProviderName()
        defer auth.SetProvider(originalAuthProvider)
	auth.SetProvider("alpha")
is better than this switch imho

Reply:
💯